main/src/ca/uwaterloo/flix/api/Flix.scala

```diff
@@ -42,7 +42,7 @@ object Flix {
 /**
   * Main programmatic interface for Flix.
   */
-class Flix {
+class Flix(testing: Boolean = false) {
```

Comment:
I think this belongs in the Options class.

Reply:
But really, now that I look through the other changes, I think we should hardcode the other classes inside the class loader.